src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -54,6 +56,8 @@ private class ConfigurationCacheItem
             _parametersCache = new MemoryCache(new MemoryCacheOptions());
         }
 
+        public ITelemetryPublisher _telemetryPublisher { get; init; }
```

Comment:
"Init" keyword is only supported by C#9.0 or later. I saw the wordaround file "IsExternalInit.cs".
I understand the "init" keyword is necessary since _telemetryPublisher is public. But I'm wondering the reason for making _telemetryPublisher public?

Reply:
This PR will be merged with the variants PR before the v3 release. Using `init` was introduced in that PR, so I'm introducing it here as well because it seems to be the best way to handle an optional DI item. https://github.com/microsoft/FeatureManagement-Dotnet/pull/250/files#diff-4bfde10b97e5e0cc44e6c1fe6085eac30888b8e35442cc6e1821389d47e3d6e6

_telemetryPublisher is public because it needs to be initialized from the ServiceCollectionExtension https://github.com/microsoft/FeatureManagement-Dotnet/pull/259/files#diff-f6cf1bd252f48965047cb4902b12dbea7fecdc20cf791a00290c1b420a5359eeR41

We could make it internal, but I don't think the benefit is worth the inability for customers to init it on their own, in case they don't use our extension.